### PR TITLE
Added parameter for model_format in models/xgboost

### LINKED ISSUE #6626: [FR] Add argument (e.g. `save_format`) to specify model save format to `mlflow.xgboost.log_model`
### Willingness to contribute

No. I cannot contribute this feature at this time.

### Proposal Summary

The XGBoost model file name is currently hardcoded as `model.xgb`:

https://github.com/mlflow/mlflow/blob/cf96d39d81932423d70d267f8a4ab6640e14ec7e/mlflow/xgboost/__init__.py#L154

This makes it impossible to [save an XGBoost model in JSON format](https://xgboost.readthedocs.io/en/stable/tutorials/saving_model.html). A new argument to specify saving format should be added and the line above should be fixed as follows:

```diff
- model_data_subpath = "model.xgb" 
+ model_data_subpath = f"model.{save_format}" 
```

References:

- https://xgboost.readthedocs.io/en/stable/tutorials/saving_model.html
- https://xgboost.readthedocs.io/en/stable/python/python_api.html#xgboost.XGBClassifier.save_model

### Motivation

> #### What is the use case for this feature?

To provide a way to save an XGBoost model in JSON format.

> #### Why is this use case valuable to support for MLflow users in general?

> #### Why is this use case valuable to support for your project(s) or organization?

> #### Why is it currently difficult to achieve this use case?

### Details

_No response_

### What component(s) does this bug affect?

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [X] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

### What interface(s) does this bug affect?

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

### What language(s) does this bug affect?

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

### What integration(s) does this bug affect?

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

### PR DESCRIPTION
**Pull request by AvikantSrivastava**
Signed-off-by: avikantsrivastava <asdavikant@gmail.com>

## Related Issues/PRs

Resolve #6626

## What changes are proposed in this pull request?

- I have added a new parameter called `model_format` in `mlflow.xgboost.save_model` and `mlflow.xgboost.log_model` which specifies the save format of xgboost model (json or xgb).
- I have passed it as a **kwargs to `Model.log` so it will be automatically picked up by `save_model` as a param
- The value defaults to `.xgb` incase it is not provided

## How is this patch tested?

- [ ] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

Still testing on examples

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [x] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@AvikantSrivastava Thanks for the PR, can we add a test in `tests/xgboost/test_xgboost_model_export.py`?

**Review by fernandosm — COMMENTED** [review]
I created a model with categorical features and tested the solution. Both `json` and `ubj` formats were correctly logged, solving the [xgboost.core.XGBoostError](https://github.com/mlflow/mlflow/issues/6626#issuecomment-1241036666):
> Check failed: !HasCategoricalSplit(): Please use JSON/UBJSON for saving models with categorical splits.

The commit LGTM.

File: mlflow/xgboost/__init__.py [hunk 1]
```diff
@@ -219,6 +222,7 @@ def log_model(
     await_registration_for=DEFAULT_AWAIT_MAX_SLEEP_SECONDS,
     pip_requirements=None,
     extra_pip_requirements=None,
+    model_format=None,
```
Comment (harupy) [inline mlflow/xgboost/__init__.py:225]: Moved the argument here to avoid a breaking change.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**